src/challenges.ts

```diff
@@ -300,3 +292,45 @@ export function getDifficultyName(difficulty: ChallengeDifficulty): string {
       return 'Unknown';
   }
 }
+
+/**
+ * Assigns the user a role based on their difficulty level.
+ * @param guild - The Discord guild where the roles are located.
+ * @param discordId - The user's Discord ID.
+ * @param difficulty - The difficulty enum for the user.
+ */
+export async function assignSageRole(
```

Comment:
This is probably generic enough to create a `assignDiscordrole`  in `discord/actions`